synapse/replication/http/federation.py

```diff
@@ -95,12 +99,13 @@ def _serialize_payload(store, event_and_contexts, backfilled):
             event_and_contexts = []
             for event_payload in event_payloads:
                 event_dict = event_payload["event"]
-                format_ver = event_payload["event_format_version"]
+                room_ver = KNOWN_ROOM_VERSIONS[event_payload["room_version"]]
```

Comment:
An unknown room version would stop processing of all events, is that expected or should we guard against that?

Reply:
I think it's reasonable. 

This is handling events sent between worker instances (specifically from the federation reader to the master) over the internal replication API. An unknown room version here could only happen if the federation reader and master were at different versions of the codebase; we generally don't make any attempt to deal with that and I think rejecting the entire request is ok.

Reply:
That seems reasonable.